src/gt4py/next/program_processors/runners/dace_fieldview/gtir_scan_translator.py

```diff
@@ -0,0 +1,678 @@
+
+    # in case tuples are passed as argument, isolated access nodes might be left in the state,
+    # because not all tuple fields are necessarily accessed inside the lambda scope
+    for data_node in compute_state.data_nodes():
+        data_desc = data_node.desc(nsdfg)
+        if compute_state.degree(data_node) == 0:
+            # By construction there should never be isolated transient nodes.
+            # Therefore, the assert below implements a sanity check, that allows
+            # the exceptional case (encountered in one GT4Py test) where the carry
+            # variable is not used, so not a scan indeed because no data dependency.
+            assert (not data_desc.transient) or data_node.data.startswith(scan_carry_symbol.id)
+            compute_state.remove_node(data_node)
+
+    return nsdfg, lambda_output
+
+
+def _connect_nested_sdfg_output_to_temporaries(
+    sdfg: dace.SDFG,
+    nsdfg: dace.SDFG,
+    nsdfg_node: dace.nodes.NestedSDFG,
+    outer_state: dace.SDFGState,
+    inner_data: gtir_translators.FieldopData,
+) -> gtir_dataflow.DataflowOutputEdge:
+    """
+    Helper function to create the edges to write output data from the nested SDFG
+    to temporary arrays in the parent SDFG, denoted as outer context.
+
+    Args:
+        sdfg: The SDFG representing the outer context, where the field operator is translated.
+        nsdfg: The SDFG where the scan `LoopRegion` is translated.
+        nsdfg_node: The nested SDFG node in the outer context.
+        outer_state: The state in outer context where the field operator is translated.
+        inner_data: The data produced by the scan `LoopRegion` in the inner context.
+
+    Returns:
+        An object representing the output data connection of this field operator.
+    """
+    assert isinstance(inner_data.gt_type, ts.FieldType)
+    inner_dataname = inner_data.dc_node.data
+    inner_desc = nsdfg.data(inner_dataname)
+    outer_dataname, outer_desc = sdfg.add_temp_transient_like(inner_desc)
+    outer_node = outer_state.add_access(outer_dataname)
+    outer_state.add_edge(
+        nsdfg_node,
+        inner_dataname,
+        outer_node,
+        None,
+        dace.Memlet.from_array(outer_dataname, outer_desc),
+    )
+    output_expr = gtir_dataflow.ValueExpr(outer_node, inner_data.gt_type.dtype)
+    return gtir_dataflow.DataflowOutputEdge(outer_state, output_expr)
+
+
+def translate_scan(
+    node: gtir.Node,
+    sdfg: dace.SDFG,
+    state: dace.SDFGState,
+    sdfg_builder: gtir_sdfg.SDFGBuilder,
+) -> gtir_translators.FieldopResult:
+    """
+    Generates the dataflow subgraph for the `as_fieldop` builtin with a scan operator.
+
+    It differs from `translate_as_fieldop()` in that the horizontal domain is lowered
+    to a map scope while the vertical dimension where the scan column is computed
+    is lowered to a `LoopRegion`. The current design choice is to keep the map scope
```

Comment:
```suggestion
    is lowered to a `LoopRegion`, that is inside the horizontal map. The current design choice is to keep the map scope
```
I think this information is important as it confused me for a long time.